src/Features/CSharp/Portable/CodeRefactorings/NodeSelectionHelpers.cs

```diff
@@ -18,8 +19,30 @@ internal static async Task<SyntaxNode> GetSelectedDeclarationOrVariableAsync(Cod
             // MemberDeclaration: member that can be declared in type (those are the ones we can pull up) 
             // VariableDeclaratorSyntax: for fields the MemberDeclaration can actually represent multiple declarations, e.g. `int a = 0, b = 1;`.
             // ..Since the user might want to select & pull up only one of them (e.g. `int a = 0, [|b = 1|];` we also look for closest VariableDeclaratorSyntax.
-            return await context.TryGetRelevantNodeAsync<MemberDeclarationSyntax>().ConfigureAwait(false) as SyntaxNode ??
-                await context.TryGetRelevantNodeAsync<VariableDeclaratorSyntax>().ConfigureAwait(false);
+            var memberDeclaration = await context.TryGetRelevantNodeAsync<MemberDeclarationSyntax>().ConfigureAwait(false) as SyntaxNode;
+            if (memberDeclaration == null)
+            {
+                return await context.TryGetRelevantNodeAsync<VariableDeclaratorSyntax>().ConfigureAwait(false);
```

Comment:
Based on the previous code, which used a null coalescing operator to first look for a `MemberDeclarationSyntax` and then a `VariableDeclaratorSyntax`, I think we can. I haven't run tests without it/to check if its hit, but based on previous behavior, selecting the variable name of a declared field wouldn't find a `FieldDeclarationSyntax`, but would find the corresponding `VariableDeclaratorSyntax`. Otherwise field member selection wouldn't have worked at all, since the refactoring providers just take the result of this helper and attempt to get the declared symbol.

Reply:
After running a debug test, this branch is indeed hit when something like `public static int F$$oo;` is selected. The `TryGetRelevantNodeAsync` does not find a `MemberDeclarationSyntax` there, but it does find a `VariableDeclaratorSyntax`